static/js/components/FollowupRequestForm.jsx

```diff
@@ -27,21 +32,32 @@ const FollowupRequestForm = ({
   const { allocationList } = useSelector((state) => state.allocations);
   const allGroups = useSelector((state) => state.groups.all);
   const [selectedAllocationId, setSelectedAllocationId] = useState(null);
+  const [selectedGroupIds, setSelectedGroupIds] = useState([]);
 
   useEffect(() => {
     // Initialize the select
     setSelectedAllocationId(allocationList[0]?.id);
-  }, [setSelectedAllocationId, allocationList]);
+    setSelectedGroupIds([allocationList[0]?.group_id]);
+  }, [setSelectedAllocationId, allocationList, setSelectedGroupIds]);
 
   // need to check both of these conditions as selectedAllocationId is
   // initialized to be null and useEffect is not called on the first
   // render to update it, so it can be null even if allocationList is not
   // empty.
-  if (allocationList.length === 0 || !selectedAllocationId) {
+  if (
+    allocationList.length === 0 ||
+    !selectedAllocationId ||
+    Object.keys(instrumentFormParams).length === 0
+  ) {
     return <h3>No robotic instruments available...</h3>;
   }
 
-  if (!allGroups || allGroups.length === 0 || telescopeList.length === 0) {
+  if (
+    !allGroups ||
+    allGroups.length === 0 ||
+    telescopeList.length === 0 ||
+    instrumentList.length === 0
```

Comment:
Optional chaining for all these?

Reply:
`telescopeList` and `instrumentList` don't need it, they are both initialized to be `[]`.  `allGroups` I have modified.

Reply:
actually need to keep it the way it is now. We want to check if `allGroups` is null or has length zero if nonnull, and the chaining wouldn't handle that. `allGroups.length` is only accessed if `!!allGroups` (previous line), so no need for chaining.